lib/PrintAsClang/PrintAsClang.cpp

```diff
@@ -309,14 +309,13 @@ static void collectClangModuleHeaderIncludes(
     requiredTextualIncludes.insert(textualInclude);
   };
 
-  if (clang::Module::Header umbrellaHeader = clangModule->getUmbrellaHeader()) {
-    addHeader(umbrellaHeader.Entry->tryGetRealPathName(),
-              umbrellaHeader.PathRelativeToRootModuleDirectory);
-  } else if (clang::Module::DirectoryName umbrellaDir =
-                 clangModule->getUmbrellaDir()) {
+  if (llvm::Optional<clang::Module::Header> umbrellaHeader = clangModule->getUmbrellaHeaderAsWritten()) {
+    addHeader(umbrellaHeader->Entry.getFileEntry().tryGetRealPathName(),
+        umbrellaHeader->PathRelativeToRootModuleDirectory);
+  } else if (llvm::Optional<clang::Module::DirectoryName> umbrellaDir = clangModule->getUmbrellaDirAsWritten()) {
     SmallString<128> nativeUmbrellaDirPath;
     std::error_code errorCode;
-    llvm::sys::path::native(umbrellaDir.Entry->getName(),
+    llvm::sys::path::native(umbrellaDir->Entry.getDirEntry().getName(),
```

Comment:
It's not clear to me what paths we actually want here, but it would be better to use `getName`/`getNameAsRequested` on the `FileEntryRef`/`DirectoryEntryRef`. `getName` on the entry will eventually disappear. We could do that in main after ranch is merged though.

CC @jansvoboda11

Reply:
Using `DirectoryEntryRef::getNameAsRequested()` here should be fine, since that path is used to query the VFS again.

Reply:
Ah, GitHub doesn't show that the comment was for the entire range very well. It's more the `getFileEntry().tryGetRealPathName()` above that I was unsure about.

Reply:
I see a `DirectoryEntryRef::getName`, but no `DirectoryEntryRef::getNameAsRequested`.
And how would you like me to resolve the path name? I don't see a replacement.

Reply:
Just `getName` on `DirectoryEntryRef` is fine (they'd be the same thing, there is no external vs request difference).

For the file it's more that `tryGetRealPathName` seems very dubious to me, since it would be ignoring the VFS when doing the include search that it seems to be doing in `addHeader`. I don't think this should block this PR, we just need to track fixing it.